src/shared/context/SwappingContext/SwappingProvider.tsx

```diff
@@ -0,0 +1,403 @@
+  allowance: null,
+  isCheckingAllowance: false,
+  isApproving: false,
+  approvalTxHash: null,
+  poolAddress: UNISWAP_POOL_ADDRESS,
+  poolFee: DEFAULT_POOL_FEE,
+}
+
+const swapReducer = (state: SwapState, action: SwapAction): SwapState => {
+  switch (action.type) {
+    case SwapActionType.SET_TOKEN_IN:
+      return {
+        ...state,
+        tokenIn: action.payload,
+        amountIn: '',
+        amountOut: null,
+        quote: null,
+        quoteError: null,
+      }
+    case SwapActionType.SET_TOKEN_OUT:
+      return {
+        ...state,
+        tokenOut: action.payload,
+        amountIn: '',
+        amountOut: null,
+        quote: null,
+        quoteError: null,
+      }
+    case SwapActionType.SET_AMOUNT_IN:
+      return {
+        ...state,
+        amountIn: action.payload,
+        amountOut: null,
+        quote: null,
+        quoteError: null,
+      }
+    case SwapActionType.SET_AMOUNT_OUT:
+      return {
+        ...state,
+        amountOut: action.payload,
+      }
+    case SwapActionType.SET_QUOTE:
+      return {
+        ...state,
+        quote: action.payload,
+        quoteError: null,
+      }
+    case SwapActionType.SET_QUOTING:
+      return {
+        ...state,
+        isQuoting: action.payload,
+      }
+    case SwapActionType.SET_QUOTE_ERROR:
+      return {
+        ...state,
+        quoteError: action.payload,
+        isQuoting: false,
+      }
+    case SwapActionType.SET_SWAPPING:
+      return {
+        ...state,
+        isSwapping: action.payload,
+      }
+    case SwapActionType.SET_SWAP_ERROR:
+      return {
+        ...state,
+        swapError: action.payload,
+        isSwapping: false,
+      }
+    case SwapActionType.SET_SWAP_TX_HASH:
+      return {
+        ...state,
+        swapTxHash: action.payload,
+      }
+    case SwapActionType.SET_ALLOWANCE:
+      return {
+        ...state,
+        allowance: action.payload,
+      }
+    case SwapActionType.SET_CHECKING_ALLOWANCE:
+      return {
+        ...state,
+        isCheckingAllowance: action.payload,
+      }
+    case SwapActionType.SET_APPROVING:
+      return {
+        ...state,
+        isApproving: action.payload,
+      }
+    case SwapActionType.SET_APPROVAL_TX_HASH:
+      return {
+        ...state,
+        approvalTxHash: action.payload,
+      }
+    case SwapActionType.SET_POOL_ADDRESS:
+      return {
+        ...state,
+        poolAddress: action.payload,
+      }
+    case SwapActionType.SET_POOL_FEE:
+      return {
+        ...state,
+        poolFee: action.payload,
+      }
+    case SwapActionType.RESET_SWAP:
+      return {
+        ...initialState,
+        tokenIn: state.tokenIn,
+        tokenOut: state.tokenOut,
+        poolAddress: state.poolAddress,
+        poolFee: state.poolFee,
+      }
+    default:
+      return state
+  }
+}
+
+interface SwappingProviderProps {
+  children: ReactNode
+}
+
+export const SwappingProvider: FC<SwappingProviderProps> = ({ children }) => {
```

Comment:
Why reducers and context providers!! We have beautiful "Zustand" installed in our app, it could do the same thing but in much much shorter way.